src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts

```diff
@@ -334,6 +320,61 @@ export class BuyCryptoService {
       }
     }
 
+    // buy update
+    if (dto.buyId) {
+      if (
+        [
+          BuyCryptoStatus.BATCHED,
+          BuyCryptoStatus.READY_FOR_PAYOUT,
+          BuyCryptoStatus.PAYING_OUT,
+          BuyCryptoStatus.COMPLETE,
+        ].includes(entity.status)
+      )
+        throw new BadRequestException('Cannot update buy route: payout already in progress or completed');
+      if (!entity.buy) throw new BadRequestException(`Cannot update buy route: entity is not a buy`);
+
+      const update: Partial<BuyCrypto> = {};
+
+      update.buy = await this.getBuy(dto.buyId);
+      if (update.buy.userData.id !== entity.transaction.userData.id)
+        throw new BadRequestException('Cannot update buy route: UserData mismatch');
+
+      // update/reset fields
+      update.outputAsset = update.buy.asset;
+      update.outputReferenceAsset = update.buy.asset;
+
+      update.percentFee = null;
+      update.percentFeeAmount = null;
+      update.minFeeAmount = null;
+      update.minFeeAmountFiat = null;
+      update.totalFeeAmount = null;
+      update.totalFeeAmountChf = null;
+      update.blockchainFee = null;
+      update.bankFeeAmount = null;
+      update.inputReferenceAmountMinusFee = null;
+      update.usedRef = null;
+      update.refProvision = null;
+      update.refFactor = null;
+      update.usedFees = null;
+      update.networkStartFeeAmount = null;
+      update.status = null;
+
+      await this.buyCryptoRepo.save(Object.assign(entity, update));
+
+      await this.buyCryptoRepo.updateFee(entity.fee.id, { feeReferenceAsset: update.buy.asset });
+
+      entity.transaction = await this.transactionService.updateInternal(entity.transaction, { user: update.buy.user });
+
+      if (entity.bankTx) await this.bankTxService.getBankTxRepo().setNewUpdateTime(entity.bankTx.id);
+    }
+
+    // crypto route update
+    if (dto.cryptoRouteId) {
+      if (!entity.cryptoRoute) throw new BadRequestException(`Cannot update crypto route: entity is not a crypto`);
+      update.cryptoRoute = await this.getCryptoRoute(dto.cryptoRouteId);
+      if (entity.bankTx) await this.bankTxService.getBankTxRepo().setNewUpdateTime(entity.bankTx.id);
```

Comment:
Why aren't there the same checks and changes as in buyRoute? E.g. check if userData would change with cryptoRoute Update, resetting the entity cols, updating transaction?